src/EnergyPlus/VariableSpeedCoils.hh

```diff
@@ -370,168 +365,11 @@ namespace VariableSpeedCoils {
-			HPCompressorRuntime( 0.0 ),
-			HPCompressorRuntimeLast( 0.0 ),
-			TimeLeftToDefrost( 0.0 ),
-			DefrostPower( 0.0 ),
-			DefrostConsumption( 0.0 ),
-			ReportCoolingCoilCrankcasePower( true ),
-			CrankcaseHeaterCapacity( 0.0 ),
-			CrankcaseHeaterPower( 0.0 ),
-			MaxOATCrankcaseHeater( 0.0 ),
-			CrankcaseHeaterConsumption( 0.0 ),
-			CondenserInletNodeNum( 0 ),
-			CondenserType( AirCooled ),
-			ReportEvapCondVars( false ),
-			EvapCondPumpElecNomPower( 0.0 ),
-			EvapCondPumpElecPower( 0.0 ),
-			EvapWaterConsumpRate( 0.0 ),
-			EvapCondPumpElecConsumption( 0.0 ),
-			EvapWaterConsump( 0.0 ),
-			BasinHeaterConsumption( 0.0 ),
-			BasinHeaterPowerFTempDiff( 0.0 ),
-			BasinHeaterSetPointTemp( 0.0 ),
-			BasinHeaterPower( 0.0 ),
-			BasinHeaterSchedulePtr( 0 ),
-			EvapCondAirFlow( MaxSpedLevels, 0.0 ),
-			EvapCondEffect( MaxSpedLevels, 0.0 ),
-			MSRatedEvapCondVolFlowPerRatedTotCap( MaxSpedLevels, 0.0 ),
-			EvapWaterSupplyMode( WaterSupplyFromMains ),
-			EvapWaterSupTankID( 0 ),
-			EvapWaterTankDemandARRID( 0 ),
-			CondensateCollectMode( CondensateDiscarded ),
-			CondensateTankID( 0 ),
-			CondensateTankSupplyARRID( 0 ),
-			CondensateVdot( 0.0 ),
-			CondensateVol( 0.0 ),
-			CondInletTemp( 0.0 ),
-			SourceAirMassFlowRate( 0.0 ),
-			InletSourceAirTemp( 0.0 ),
-			InletSourceAirEnthalpy( 0.0 ),
-			//begin varibles for HPWH
-			RatedCapWH( 0.0 ), // Rated water heating Capacity [W]
-			InletAirTemperatureType( 0 ), // Specifies to use either air wet-bulb or dry-bulb temp for curve objects
-			WHRatedInletDBTemp( 0.0 ), // Rated inlet air dry-bulb temperature [C]
-			WHRatedInletWBTemp( 0.0 ),  // Rated inlet air wet-bulb temperature [C]
-			WHRatedInletWaterTemp( 0.0 ),  // Rated condenser water inlet temperature [C]
-			HPWHCondPumpElecNomPower( 0.0 ),  // Nominal power input to the condenser water circulation pump [W]
-			HPWHCondPumpFracToWater( 1.0 ),  // Nominal power fraction to water for the condenser water circulation pump
-			RatedHPWHCondWaterFlow( 0.0 ), // Rated water flow rate through the condenser of the HPWH DX coil [m3/s]
-			ElecWaterHeatingPower( 0.0 ),  // Total electric power consumed by compressor and condenser pump [W]
-			ElecWaterHeatingConsumption( 0.0 ),  // Total electric consumption by compressor and condenser pump [J]
-			FanPowerIncludedInCOP( false ), // Indicates that fan heat is included in heating capacity and COP
-			CondPumpHeatInCapacity( false ), // Indicates that condenser pump heat is included in heating capacity
-			CondPumpPowerInCOP( false ), // Indicates that condenser pump power is included in heating COP
-			AirVolFlowAutoSized( false ), // Used to report autosizing info for the HPWH DX coil
-			WaterVolFlowAutoSized( false ), // Used to report autosizing info for the HPWH DX coil
-			TotalHeatingEnergy( 0.0 ),  //total water heating energy
-			TotalHeatingEnergyRate( 0.0 ) //total WH energy rate
-			//end variables for HPWH
-		{}
+		VariableSpeedCoilData();
```

Comment:
the default constructor is in C++ than the struct header file,  due to needing some pre-defined constant for initialization, like int const MaxSpedLevels( 10 ); // Maximum number of speed that supports
This will make it easy to increase the upper speed level later.